### PR TITLE
Upgrade OpenAPI spec to v3

### PR DESCRIPTION
**Pull request by maukoquiroga**
#### New Features

- Upgrade OpenAPI specification of the API to v3 from Swagger v2.
- Continuously validate OpenAPI specification.

#### Breaking changes

- Drop support for OpenAPI specification v2 and prior.
- Users relying on the aforesaid can use [this](https://converter.swagger.io/api/convert?url=OAS2_YAML_OR_JSON_URL) to migrate ([example](https://web.archive.org/web/20221103230822/https://converter.swagger.io/api/convert?url=https://api.demo.openfisca.org/latest/spec)).

screenshot

### CODE REVIEW

File: CHANGELOG.md
Comment (MattiSG) [inline CHANGELOG.md:7]: - Upgrade Web API specification to OpenAPI v3.

File: CHANGELOG.md
Comment (MattiSG) [inline CHANGELOG.md:5]:
```suggestion
#### New features
```

File: CHANGELOG.md
Comment (MattiSG) [inline CHANGELOG.md:8]: Inaccurate in this package.
```suggestion
```

File: CHANGELOG.md
Comment (MattiSG) [inline CHANGELOG.md:13]:
```suggestion
  - Users relying on OpenAPI v2 can use [Swagger Converter](https://converter.swagger.io/api/convert?url=OAS2_YAML_OR_JSON_URL) to migrate ([example](https://web.archive.org/web/20221103230822/https://converter.swagger.io/api/convert?url=https://api.demo.openfisca.org/latest/spec)).
```

File: setup.py
Comment (MattiSG) [inline setup.py:13]: Update before merge.

File: setup.py
Comment (MattiSG) [inline setup.py:41]: Update before merge.

**Review by MattiSG — CHANGES_REQUESTED** [review]
Beyond the requested changes here (blocking but trivial, basically just CHANGELOG copy), this is blocked until https://github.com/openfisca/legislation-explorer/ is updated to support OpenAPI v3, as https://demo.openfisca.org will automatically deploy the latest version of this package.

**Comment by maukoquiroga** [comment]
> Beyond the requested changes here (blocking but trivial, basically just CHANGELOG copy), this is blocked until https://github.com/openfisca/legislation-explorer/ is updated to support OpenAPI v3, as https://demo.openfisca.org will automatically deploy the latest version of this package.

Legislation explorer already handles OpenAPI v3.

**Review by MattiSG — APPROVED** [review]

**Pull request merged**
